neuraxle/pipeline.py

```diff
@@ -486,3 +490,108 @@ def should_resume(self, data_container: DataContainer) -> bool:
+            data_inputs=data_inputs
+        )
+        data_container = DataContainer(current_ids=current_ids, data_inputs=data_inputs)
+        data_container = self.handle_transform(data_container)
+
+        return data_container.data_inputs
+
+    def fit_transform(self, data_inputs, expected_outputs=None) -> ('Pipeline', Any):
+        """
+        :param data_inputs: the data input to fit on
+        :param expected_outputs: the expected data output to fit on
+        :return: the pipeline itself
+        """
+        self.setup()
+
+        current_ids = self.hash(
+            current_ids=None,
+            hyperparameters=self.hyperparams,
+            data_inputs=data_inputs
+        )
+        data_container = DataContainer(
+            current_ids=current_ids,
+            data_inputs=data_inputs,
+            expected_outputs=expected_outputs
+        )
+        new_self, data_container = self.handle_fit_transform(data_container)
+
+        return new_self, data_container.data_inputs
+
+    def handle_transform(self, data_container: DataContainer) -> DataContainer:
+        """
+        Transform all sub pipelines splitted by the Barrier steps.
+
+        :param data_container: data container to transform.
+        :return: data container
+        """
+        sub_pipelines = self._create_sub_pipelines()
+
+        for sub_pipeline in sub_pipelines:
+            barrier = sub_pipeline[-1]
+            data_container = barrier.join_transform(
+                step=sub_pipeline,
+                data_container=data_container
+            )
+            current_ids = self.hash(data_container.current_ids, self.hyperparams)
+            data_container.set_current_ids(current_ids)
+
+        return data_container
+
+    def handle_fit_transform(self, data_container: DataContainer) -> \
+            Tuple['MiniBatchSequentialPipeline', DataContainer]:
+        """
+        Transform all sub pipelines splitted by the Barrier steps.
+
+        :param data_container: data container to transform.
+        :return: data container
+        """
+        sub_pipelines = self._create_sub_pipelines()
+
+        for sub_pipeline in sub_pipelines:
+            barrier = sub_pipeline[-1]
+            sub_pipeline, data_container = barrier.join_fit_transform(
+                step=sub_pipeline,
+                data_container=data_container
+            )
+            current_ids = self.hash(data_container.current_ids, self.hyperparams)
+            data_container.set_current_ids(current_ids)
+
+        return self, data_container
+
+    def _create_sub_pipelines(self) -> List['MiniBatchSequentialPipeline']:
+        """
+        Create sub pipelines by splitting the steps by the join type name.
+
+        :return: list of sub pipelines
+        """
+        sub_pipelines: List[MiniBatchSequentialPipeline] = self.split(Barrier)
+        for sub_pipeline in sub_pipelines:
+            if not sub_pipeline.ends_with(Barrier):
```

Comment:
Seems like you resolved, but just forgot the constant. ok

Reply:
it's better like this resolving.